src/v/raft/group_manager.h

```diff
@@ -99,6 +99,18 @@ class group_manager {
         return _recovery_scheduler.get_status();
     }
 
+    using remake_cb_t
+      = ss::noncopyable_function<ss::future<std::error_code>(raft::group_id)>;
+    // Remake callback is set in controller.cc to tie together `group_manager`
+    // to `controller_backend/api`.
+    void set_remake_cb(remake_cb_t cb) {
+        _remake_cb = std::make_unique<remake_cb_t>(std::move(cb));
+    }
+
+    // Lifetime of `group_manager` & `controller_backend/api` is not tied. We
+    // need to clear this callback to avoid calls to stopped services.
+    void clear_remake_cb() { _remake_cb = nullptr; }
```

Comment:
Do we need to hold a gate to ensure the callback isn't in flight when this is destructed?